Version update mp5 (#122)
OpenLiberty/draft-guide-cloud-ibm: README.adoc

== What you'll learn

You will learn how to deploy two microservices in Open Liberty containers to a {kube} cluster on IBM Cloud. 

Kubernetes is an open source container orchestrator that automates many tasks involved in deploying, managing, and scaling containerized applications. If you would like to learn more about Kubernetes, check out the https://openliberty.io/guides/kubernetes-intro.html[Deploying microservices to Kubernetes^] guide.

> There are different cloud-based solutions for running your Kubernetes workloads. A cloud-based infrastructure enables you to focus on developing your microservices without worrying about low-level infrastructure details for deployment. Using a cloud helps you to easily scale and manage your microservices in a high-availability setup.

Reviewer: ```suggestion
Different cloud-based solutions are available to run your Kubernetes workloads. A cloud-based infrastructure enables you to focus on developing your microservices without worrying about low-level infrastructure details for deployment. Using a cloud helps you to easily scale and manage your microservices in a high-availability setup.
```

> IBM Cloud Kubernetes Service (IKS) is part of IBM’s public cloud offerings. It provides a hosted Kubernetes cluster where you can deploy your microservices. You will use it with IBM Cloud Container Registry, a private registry used to store and distribute your container images.

Reviewer: ```suggestion
The IBM Cloud Kubernetes Service (IKS) is part of IBM’s public cloud offerings. It provides a hosted Kubernetes cluster where you can deploy your microservices. In this guide, you will use it with the IBM Cloud Container Registry, which is a private registry that is used to store and distribute your container images.
```

> IBM Cloud Kubernetes Service (IKS) is part of IBM’s public cloud offerings. It provides a hosted Kubernetes cluster where you can deploy your microservices. You will use it with IBM Cloud Container Registry, a private registry used to store and distribute your container images.

Reviewer: I haven't corrected it throughout this guide, but IBM terminology says not to use "IKS" as an abbreviation for "IBM Kubernetes Service"

Author: Removed the IKS

> The two microservices you will deploy are called `system` and `inventory`. The system microservice returns the JVM system properties of the running container. It also returns the pod’s name in the HTTP header, making replicas easy to distinguish from each other. The `inventory` microservice adds the properties from the `system` microservice to the inventory. This demonstrates how communication can be established between pods inside a cluster.

Reviewer: ```suggestion
The two microservices you will deploy are called `system` and `inventory`. The `system` microservice returns the JVM system properties of the running container. It also returns the pod’s name in the HTTP header, making replicas easy to distinguish from each other. The `inventory` microservice adds the properties from the `system` microservice to the inventory. This configuration demonstrates how to establish communication between pods inside a cluster.
```

> You will use Helm to deploy these microservices to IBM Cloud using Open Liberty Helm charts. Helm is a package manager for {kube}. It uses templates to generate {kube} yaml files and then deploys and manages them as releases in your cluster. 

Reviewer: ```suggestion
You will use Helm to deploy these microservices to IBM Cloud by using Open Liberty Helm charts. Helm is a package manager for {kube}. It uses templates to generate {kube} YAML files and then deploys and manages them as releases in your cluster. 
```

=== Finding the microservice's IP address and ports

> The service used to expose your deployments has a type of `NodePort`. This means you can access these services from outside of your cluster via a specific port. In this case, `nodePort` is not specified so the ports are randomized so you must obtain the ports before making requests to the services. You must also obtain the public IP address of your cluster. Note that there are other ways to expose your services such as using a `LoadBalancer` service type or using an `Ingress`. In production, you would most likely use an `Ingress`.

Reviewer: ```suggestion
The service used to expose your deployments has a type of `NodePort` service that enables you to access these services from outside of your cluster through a specific port. In this case, `nodePort` is not specified and the ports are randomized so you must obtain the ports before you make requests to the services. You must also obtain the public IP address of your cluster. Other ways to expose your services are available, such as by using a `LoadBalancer` service type or  an `Ingress` resource. In production, you typically use an `Ingress` resource.
```

ibmcloud ks workers --cluster guide-cluster
```

> Take note of the `Public IP` in the command's output. This will be the hostname you substitute into commands later in this guide.

Reviewer: ```suggestion
Take note of the `Public IP` in the command's output. You will substitute this value as the hostname into commands later in this guide.
```

=== Making requests to the microservices

To make a request to the `system` and `inventory` microservices,  `curl` or visit the following URLs to access your microservices, substituting the appropriate hostname and node ports:

* `{system-api}`
* `{inventory-api}/system-service`

> The first URL returns system properties and the name of the pod in an HTTP header called `X-Pod-Name`. To view the header, you can use the `-I` option in the `curl` when making a request to `{system-api}`. The second URL adds properties from `system-service` to the inventory.

Reviewer: ```suggestion
The first URL returns system properties and the name of the pod in an HTTP header called `X-Pod-Name`. To view the header, you can use the `-I` option in the `curl` when making a request to  the `{system-api}` URL. The second URL adds properties from `system-service` to the inventory.
```

// =================================================================================================
// Testing microservices that are running on IBM Cloud
// =================================================================================================

== Testing microservices that are running on IBM Cloud

> A few tests are included for you to test the basic functionality of the microservices. If a test failure occurs, then you might have introduced a bug into the code. To run the tests, wait for all pods to be in the ready state before proceeding further. The default properties that are defined in the `pom.xml` are:

Reviewer: ```suggestion
A few tests are included for you to test the basic functionality of the microservices. If a test fails, then you might have introduced a bug into the code. To run the tests, wait until all pods are in the ready state before proceeding further. The default properties that are defined in the `pom.xml` file are:
```

> == Deploying new version of system microservice

Reviewer: ```suggestion
== Deploying a new version of system microservice
```

> Optionally, you might want to make changes to your microservice and learn how to redeploy the updated version of your microservice. In this section, you will bump the version of the `system` microservice to `2.0-SNAPSHOT` and redeploy the new version of the microservice. You can redeploy a microservice using Helm to upgrade the release.

Reviewer: ```suggestion
Optionally, you might want to make changes to your microservice and redeploy the updated version. In this section, you will bump the version of the `system` microservice to `2.0-SNAPSHOT` and redeploy the new version of the microservice. You can redeploy a microservice by using Helm to upgrade the release.
```

[source, role="no_copy"]
----
NAME                                       READY    STATUS    RESTARTS   AGE
inventory-app-ibm-open-l-994778cf7-fdkb5   1/1      Running   0          5m
system-app-ibm-open-libe-85c85949c-lb49d   1/1      Running   0          23s
----

> Observe that in this case the `system` microservice is running in the pod called `system-app-ibm-open-libe-85c85949c-lb49d`. Substitute the name of your pod into the following command to see more details about the pod.

Reviewer: ```suggestion
In this case, the `system` microservice is running in the pod called `system-app-ibm-open-libe-85c85949c-lb49d`. Substitute the name of your pod into the following command to see more details about the pod.
```

== Great work! You're done!

> You have just deployed two microservices to IBM Cloud. You have learned to use Helm to install your microservices on a {kube} cluster using the Open Liberty helm chart.

Reviewer: ```suggestion
You just deployed two microservices to IBM Cloud. You learned to use Helm to install your microservices on a {kube} cluster by using the Open Liberty helm chart.
```
